pennylane/qnn/torch.py

```diff
@@ -302,14 +308,14 @@ def forward(self, inputs):  # pylint: disable=arguments-differ
             tensor: output data
         """
 
-        if len(inputs.shape) > 1:
+        if self.split_batches and len(inputs.shape) > 1:
             # If the input size is not 1-dimensional, unstack the input along its first dimension,
             # recursively call the forward pass on each of the yielded tensors, and then stack the
             # outputs back into the correct shape
             reconstructor = [self.forward(x) for x in torch.unbind(inputs)]
             return torch.stack(reconstructor)
 
-        # If the input is 1-dimensional, calculate the forward pass as usual
+        # calculate the forward pass as usual
```

Comment:
The next lines are executed if either i) the input is 1-dimensional, or ii) `split_batches` is `False`, in which case it will make use of parameter broadcasting. I can add a comment to make this explicit.

Reply:
Ah, I see. Not a blocker from my side, but when in doubt, giving helpful comments doesn't hurt :)

Reply:
Added some more comments 👍